### PR TITLE
feat: Add the mender-snapshot sources

### PR DESCRIPTION
**Pull request by oleorhagen**
This adds the initial sources for the mender-snapshot tool, partially removed from the `mender-client`.

In most aspects this mirrors the `mender-setup` tool, in it's sources, and structure, and is pretty much a copy-paste, at least for the scaffolding and structure.

Ticket: MEN-6452
Changelog: None

### CODE REVIEW

**Comment by oleorhagen** [comment]
I have not tested it yet...

**Comment by oleorhagen** [comment]
I have not tested it yet...

Also, note, we have no tests here...

**Comment by lluiscampos** [comment]
> I have not tested it yet...
> 
> Also, note, we have no tests here...

Good point - this was tested in client acceptance tests. We should create a task to revisit in the future - either instrumenting this tool and collecting the coverage or using the more fancy toolset of golang 1.20

**Comment by oleorhagen** [comment]
> > I have not tested it yet...
> > Also, note, we have no tests here...
> 
> Good point - this was tested in client acceptance tests. We should create a task to revisit in the future - either instrumenting this tool and collecting the coverage or using the more fancy toolset of golang 1.20

Have the golang developers stolen my project?

**Comment by lluiscampos** [comment]
> Have the golang developers stolen my project?

[Yes](https://go.dev/testing/coverage/).

File: cli/cli.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+						Usage: "Compression type to use on the" +
+							"rootfs snapshot {none,gzip}",
+						Value: "none",
+					},
+				},
+			},
+		},
+	}
+	app.Flags = []cli.Flag{
+		&cli.StringFlag{
```
Comment (lluiscampos) [inline cli/cli.go:110]: We don't need the following flags:
```

   --fallback-config FILE, -b FILE  Fallback configuration FILE path. (default:
                                    "/var/lib/mender/mender.conf")
   --data DIR, -d DIR               Mender state data DIRECTORY path. (default:
                                    "/var/lib/mender")

   --trusted-certs FILE, -E FILE    Trusted server certificates FILE path.
   --forcebootstrap, -F             Force bootstrap. (default: false)

   --skipverify                     Skip certificate verification. (default: false)
   --passphrase-file value
```

File: cli/cli.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+			Name:        "config",
```
Comment (lluiscampos) [inline cli/cli.go]: Do we need config? I think we don't

File: cli/cli.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+			Destination: &runOptions.HttpConfig.ServerCert,
+		},
+		&cli.BoolFlag{
+			Name:        "forcebootstrap",
+			Aliases:     []string{"F"},
+			Usage:       "Force bootstrap.",
+			Destination: &runOptions.bootstrapForce,
+		},
+		&cli.BoolFlag{
+			Name:        "no-syslog",
```
Comment (lluiscampos) [inline cli/cli.go]: Does this one make sense for the snapshot?

**Review by lluiscampos — CHANGES_REQUESTED** [review]
I think we can remove many CLI options (most of the global options).

Also I didn't review these for the `mender-setup` but please consider cleaning up there also after this review :sponge:

File: cli/cli.go (hunk 1 above)
Comment (oleorhagen) [reply cli/cli.go:110]: Looks to me like we only need the `log-level` flag actually

**Review by lluiscampos — APPROVED** [review]

**Comment by oleorhagen** [comment]
@lluiscampos will you please merge it :pray:

**Pull request merged**